bindgen/codegen/mod.rs

```diff
@@ -1285,20 +1291,27 @@ impl CodeGenerator for TemplateInstantiation {
             let align_of_expr = quote! {
                 ::#prefix::mem::align_of::<#ident>()
             };
+            let size_of_err =
+                format!("Size of template specialization: {name}");
+            let align_of_err =
+                format!("Align of template specialization: {name}");
 
-            let item = quote! {
-                #[test]
-                fn #fn_name() {
-                    assert_eq!(#size_of_expr, #size,
-                               concat!("Size of template specialization: ",
-                                       stringify!(#ident)));
-                    assert_eq!(#align_of_expr, #align,
-                               concat!("Alignment of template specialization: ",
-                                       stringify!(#ident)));
-                }
-            };
-
-            result.push(item);
+            if compile_time {
+                result.push(quote! {
+                    const _: () = {
+                        assert!(#size_of_expr == #size, #size_of_err);
```

Comment:
Is there a reason to prefer the `assert!(x == y)` form for the compile-time assertions, rather than the `assert_eq!(x, y)` form in use for runtime assertions?

My quick test of `BINDGEN_OVERWRITE_EXPECTED=1 cargo test; cargo test` seemed to yield reasonable results.

Reply:
I think `assert_eq` is not available in `const` contexts yet: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2b310bd186df07db2a4acd4e04b9a1a7

Reply:
I see what you mean, @pvdrz -- I guess I must have forgotten how to actually run the generated tests (as opposed to testing the output for matching expectations).